third_party/move/move-compiler/src/parser/ast.rs

```diff
@@ -234,6 +234,30 @@ new_name!(FunctionName);
 pub const NATIVE_MODIFIER: &str = "native";
 pub const ENTRY_MODIFIER: &str = "entry";
 
+/// An access specifier describes the resources being accessed by a function.
+/// In contrast to regular `NameAccessChain`, the identifiers inside of the
+/// chain can be wildcards (`*`).
+#[derive(Debug, Clone, PartialEq)]
+pub enum AccessSpecifier_ {
+    Acquires(bool, NameAccessChain, Option<Vec<Type>>, AddressSpecifier),
```

Comment:
There are pros and cons to both approaches. Factoring out makes things more complex. Rust's pattern matching easily deals concisely with the current design pattern. I prefer to stay with this for the parser AST. As a matter of fact, the model AST and file format do what you suggest.

Reply:
I think in that case, here, even having a brief comment like this:

`// (negated, resource, instantiation, address)`

at the top would be useful, and a reasonable compromise. Right now, you have to hunt for their uses to determine their meaning: in particular, one could read the bool field as true if we acquire, eg.

Reply:
I agree with Vineeth that this is nonoptimal, as we don't get to abstract away the kind of access from the access specification.  Using Rust's pattern matching is convenient, but wordy.  Right now we have 3 clauses to match these three cases, and each clause has to mention all 4 data fields.  If the 4 data fields were grouped as a type, then processing the fields would be abstracted away.  If we decide to add more entries to Acquires, Reads, and Writes in the future (I can imagine something related to Aggregations/Reductions and PrivatizableResources) then we duplicate the processing.

BUT: This can be refactored later, this is good enough for now.

Reply:
If this gets significantly more variants over time, I agree we should refactor this.